File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    static int[] supo1 = {1, 2, 3, 4, 5};
    static int[] supo2 = {2, 1, 2, 3, 2, 4, 2, 5};
    static int[] supo3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    public int[] solution(int[] answers) {
        int[] corrects = new int[3];
        
        for (int i = 0; i < answers.length; i++) {
            int guess1 = supo1[i % supo1.length];
            int guess2 = supo2[i % supo2.length];
            int guess3 = supo3[i % supo3.length];
            int answer = answers[i];
            
            if (guess1 == answer) {
                corrects[0]++;
            }
            if (guess2 == answer) {
                corrects[1]++;
            }
            if (guess3 == answer) {
                corrects[2]++;
            }
        }
        
        int max = Arrays.stream(corrects).max().getAsInt();
        List<Integer> list = new ArrayList<>();
        for (int i = 0; i < corrects.length; i++) {
            if (max == corrects[i]) {
                list.add(i + 1);
            }
        }
        
        int[] result = new int[list.size()];
        int index = 0;
        for (int num : list) {
            result[index++] = num;
        }
        return result;
    }
}

File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    ways = [
        [1, 2, 3, 4, 5],
        [2, 1, 2, 3, 2, 4, 2, 5],
        [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    ]
    
    pts = [0] * 3
    for i, ans in enumerate(answers):
        for j, way in enumerate(ways):
            if ans == way[i % len(way)]:
                pts[j] += 1
    
    max_pt = max(pts)
    ret = []
    for i, pt in enumerate(pts):
        if pt == max_pt:
            ret.append(i + 1)

    return ret